packages/deployer/subtasks/deploy-router.js

```diff
@@ -1,15 +1,22 @@
+const path = require('path');
 const logger = require('../utils/logger');
 const { subtask } = require('hardhat/config');
 const { TASK_COMPILE } = require('hardhat/builtin-tasks/task-names');
+const { getRouterName } = require('../utils/router');
+const relativePath = require('../utils/relative-path');
 
 const { SUBTASK_DEPLOY_CONTRACT, SUBTASK_DEPLOY_ROUTER } = require('../task-names');
 
-subtask(SUBTASK_DEPLOY_ROUTER).setAction(async (_, hre) => {
+subtask(SUBTASK_DEPLOY_ROUTER).setAction(async ({ instance }, hre) => {
   logger.subtitle('Deploying router');
 
   await hre.run(TASK_COMPILE, { force: false, quiet: true });
 
-  const contractPath = hre.deployer.paths.router;
+  const contractPath = path.join(
+    relativePath(hre.config.paths.sources),
+    getRouterName({ network: hre.network.name, instance })
```

Comment:
maybe you could create an util named getRouterPath, and recieves the instance, network & hre.config.paths.sources as params. And this util could be used in other places that need the router path

Reply:
I thought about this, but concluded that it wouldn't be worth the trouble. Let's leave it for now.